### DISCUSSION TITLE
How to run custom USD/Hydra render delegate in newly released Maya 2024

### DISCUSSION THREAD

**Discussion by Vsevolod1983** [Autodesk/maya-usd#2984] [General]
Hello!

Previously on Maya 2023 we were managed to run our own custom hydra render delegate with MtoH/USD ecosystem which was part of Maya installation

I tried newly released Maya 2024 and I found out that MtoH is absent and I cannot see our render delegate in the list of available renderers in viewport.

Previously, we used Maya.env file to have proper setup :
PATH=%PATH%; < Path to libraries of custom render delegate>
PXR_PLUGINPATH_NAME=%PXR_PLUGINPATH_NAME%; <Path to Hydra Plugin for custom render delegate>

It seems that this approach does not work any more. 
Could you please suggest how we should work with it now in Maya 2024 ?

**Comment by lanierd-adsk** [comment]
Hi,
The method still remains the same using PXR_PLUGINPATH_NAME, but we renamed the mtoh plugin to mayahydra.mll which must be loaded using the plugin manager.
Hope this helps.
Regards,
David

**Reply by Vsevolod1983** [reply]
@lanierd 
Hi ! Thank you for your response.

It seems to be that mayaHydra is a part of mayaUsd 0.23.0. But Maya 2024 has 0.22.0 version inside.
So, we don't have it with Maya 2024.
Is it correct ?

**Reply by bjorn-siegert-adsk** [reply]
Yes, it is correct that we removed mtoh and mayaHydra is not present in mayaUSD 0.22.0 and therefore in the Maya 2024 installer. But mayaHydra is available from mayaUSD 0.23.0 onwards. 
One thing I want to add is that mayaHydra is not just a renaming of the plugin. We are continuing the work done with MtoH to make Hydra an integrated part in Maya as a viewport. Some more details are in the readme [here](https://github.com/Autodesk/maya-usd/blob/release/maya-hydra/lib/mayaHydra/doc/mayaHydraDetails.md)

**Reply by Vsevolod1983** [reply]
Ok, thank you!

**Reply by gaddipvi** [reply]
we are not getting mayaHydra in version 0.25.0 for maya 2023 version on Linux. Any tricks to get this working?

**Reply by bjorn-siegert-adsk** [reply]
@gaddipvi mayaHydra requires at least Maya 2024. If you are referring to the previous experimental MtoH plugin then you can re-enable it in the mayaUSD build manually.
